src/components/ApplicationGroup/AppGroup.service.ts

```diff
@@ -68,6 +70,29 @@ export const getWorkflows = (envID: string): Promise<WorkflowsResponseType> => {
     })
 }
 
+export const getCIConfigList = (envID: string): Promise<any> => {
+    const pipelineList = []
+    const _filteredworkflow: Map<string, any> = new Map()
+    return Promise.all([
+        getEnvWorkflowList(envID),
+        getCIConfig(envID),
+        getModuleInfo(ModuleNameMap.SECURITY),
+        getModuleConfigured(ModuleNameMap.BLOB_STORAGE),
+    ]).then(([workflow, ciConfig, securityInfo, moduleConfig]) => {
+        workflow.result.workflows.forEach((workFlow) => {
+            const selectedTree = workFlow.tree.find((list) => list.type === PipelineType.CI_PIPELINE)
+            _filteredworkflow.set(workFlow.appId, selectedTree)
+        })
+
+        ciConfig.result.forEach((item) => {
+            let ciPipeline = _filteredworkflow.get(item.appId)
+            let pipelineData = item.ciPipelines.find((pipeline) => pipeline.id === ciPipeline?.componentId)
```

Comment:
incase of ciPipeline is undefined the pipelineData will be undefined. Which could result in error. we should handle that